File: Python3/프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
import collections

def solution(queue1, queue2):
    queue1 = collections.deque(queue1)
    queue2 = collections.deque(queue2)
    total = len(queue1) + len(queue2)
    cnt = 0
    
    sum_1 = sum(queue1)
    sum_2 = sum(queue2)
    
    while cnt < (total * 4):
        # print(queue1, queue2)
        
        if sum_1 == sum_2:
            return cnt
        elif sum_1 > sum_2:
            temp = queue1.popleft()
            queue2.append(temp)
            sum_1 -= temp
            sum_2 += temp
        elif sum_1 < sum_2:
            temp = queue2.popleft()
            queue1.append(temp)
            sum_2 -= temp
            sum_1 += temp
        
        cnt += 1
            
    return -1

File: Swift/프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.swift
import Foundation

func solution(_ queue1:[Int], _ queue2:[Int]) -> Int {
    var q1 = Queue()
    q1.elements = queue1
    var sum1 = queue1.reduce(0,+)
    
    var q2 = Queue()
    q2.elements = queue2
    var sum2 = queue2.reduce(0,+)
    
    var total = (queue1.count + queue2.count) * 2
    var cnt = 0
    
    while sum1 != sum2 && cnt <= total {
        cnt += 1
        if sum1 > sum2 {
            var temp = q1.popLeft()
            q2.append(temp)
            
            sum1 -= temp
            sum2 += temp
        } else {
            var temp = q2.popLeft()
            q1.append(temp)
            
            sum2 -= temp
            sum1 += temp
        }
    }
    
    if sum1 == sum2 {
        return cnt
    } else {
        return -1
    }
}

class Queue {
    var elements = [Int]()
    var head = 0
    
    func popLeft() -> Int {
        var temp = elements[head]
        head += 1
        return temp
    }
    
    func append(_ data: Int) {
        elements.append(data)
    }
}